Achievements/SelfMade.lua

```diff
@@ -57,35 +55,34 @@ local function isSelfCreated(...)
                     return true
                 end
             end
-            if i >= 18 then -- this ensures that the loop ends
-                if text == nil then
-                    GameTooltip:Hide() --prevents a hung empty tooltip window
-                    return false
-                end
-            end
         end
     end
-
+    GameTooltip:Hide() --prevents a hung empty tooltip window
+    return false
 end
 
 -- Register Definitions
 self_made_achievement:SetScript("OnEvent", function(self, event, ...)
     GameTooltip:SetOwner(WorldFrame, "ANCHOR_CURSOR") -- Anchors the cursor so it ensure we check the right tooltip
     local arg = { ... }
     if event == "PLAYER_EQUIPMENT_CHANGED" then
```

Comment:
Is isSelfCreated supposed to be called if PLAYER_EQUIPMENT_CHANGED is triggered?

Reply:
> Is isSelfCreated supposed to be called if PLAYER_EQUIPMENT_CHANGED is triggered?

It has to because that is the event that triggers when someone swaps gear, so we grab the tooltip of the gear they are putting on, then check the information to ensure it has the made by tag and it matches their player name. The original iteration that we started with had that same logic in it. The other event ITEM_UNLOCKED is for the bags only.

Reply:
Ah, i dont think its being called here, looks like its just called for item unlocked

Reply:
> Ah, i dont think its being called here, looks like its just called for item unlocked

So i think i see what your talking about? Your pointing out the if elseif logic? Where you dont think its actually making it to the code when the PLAYER_EQUIPMENT_CHANGED? I see the concern now. Ill do some playtesting to see what portion os firing when i get home today.